packages/component-boilerplate/src/environment/environment.ts

```diff
@@ -0,0 +1,257 @@
+      android: SquatchAndroid;
+      widgetIdent: WidgetIdent;
+    }
+  | {
+      type: "SquatchPortal";
+      context: SquatchPortal;
+      env: PortalEnv;
+    }
+  | {
+      type: "SquatchAdmin";
+      adminSDK: SquatchAdmin;
+    }
+  | {
+      type: "None";
+    };
+
+/**
+ * Get the type of environment that this widget is being rendered in
+ *
+ * Should never return null.
+ */
+export function getEnvironment(): Environment {
+  return getEnvironmentSDK().type;
+}
+
+/**
+ * Get the SDK for interacting with the host environment
+ */
+export function getEnvironmentSDK(): EnvironmentSDK {
+  //@ts-ignore
+  if (window["SquatchAndroid"]) {
+    return {
+      type: "SquatchAndroid",
+      //@ts-ignore
+      android: window["SquatchAndroid"] as SquatchAndroid,
+      //@ts-ignore
+      widgetIdent: window["widgetIdent"],
+    };
+  }
+
+  //@ts-ignore
+  if (window["SquatchPortal"]) {
+    return {
+      type: "SquatchPortal",
+      context: SquatchPortalInstance,
+      //@ts-ignore
+      env: window["SquatchPortal"],
+    };
+  }
+
+  //@ts-ignore
+  if (window["SquatchAdmin"]) {
+    return {
+      type: "SquatchAdmin",
+      //@ts-ignore
+      adminSDK: window["SquatchAdmin"] as SquatchAdmin,
+    };
+  }
+
+  if (window["widgetIdent"]) {
+    return {
+      type: "SquatchJS2",
+      //@ts-ignore
+      api: window.frameElement?.["squatchJsApi"],
+      //@ts-ignore
+      widgetIdent: window["widgetIdent"],
+    };
+  }
+
+  return {
+    type: "None",
+  };
+}
+
+export function isDemo() {
+  const sdk = getEnvironmentSDK();
+  return sdk.type === "None" || sdk.type === "SquatchAdmin";
+}
+
+// Fake tenant alias in
+const FAKE_TENANT = "demo";
+
+export function useTenantAlias(): string {
+  const sdk = getEnvironmentSDK();
+  switch (sdk.type) {
+    case "SquatchAndroid":
+    case "SquatchJS2":
+      return sdk.widgetIdent.tenantAlias;
+    case "SquatchAdmin":
+    case "None":
+      return FAKE_TENANT;
+    case "SquatchPortal":
+      return sdk.env.tenantAlias;
+  }
+}
+
+const DEFAULT_DOMAIN = "https://app.referralsaasquatch.com";
+export function useAppDomain(): string {
+  const sdk = getEnvironmentSDK();
+  switch (sdk.type) {
+    case "SquatchAndroid":
+    case "SquatchJS2":
+      return sdk.widgetIdent.appDomain;
+    case "SquatchPortal":
+      return sdk.env?.appDomain || DEFAULT_DOMAIN;
+    case "SquatchAdmin":
+    case "None":
+      return DEFAULT_DOMAIN;
+  }
+}
+
+export type UserId = {
+  id: string;
+  accountId: string;
+};
+
+type EngagementMedium = "EMBED" | "POPUP";
+const DEFAULT_MEDIUM = "EMBED";
+
+export function useEngagementMedium(): EngagementMedium {
+  const sdk = getEnvironmentSDK();
+  switch (sdk.type) {
+    case "SquatchJS2":
+      return sdk.widgetIdent.engagementMedium || DEFAULT_MEDIUM;
+    case "SquatchAndroid":
+    case "SquatchPortal":
+    case "SquatchAdmin":
+    case "None":
+      return DEFAULT_MEDIUM;
+  }
+}
+
+export function useLocale(): string {
```

Comment:
@sam-beveridge is this used?

Reply:
doesn't look like we use anything Locale related, we can probably scrap it for now and implement when its needed